presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java

```diff
@@ -99,7 +107,8 @@ private void checkNoSpillInProgress()
 
     private void writePages(Iterator<Page> pageIterator)
     {
-        try (SliceOutput output = new OutputStreamSliceOutput(new FileOutputStream(targetFileName.toFile(), true))) {
+        try (SliceOutput output = new OutputStreamSliceOutput(new FileOutputStream(targetFileName.toFile(), true), BUFFER_SIZE)) {
+            memoryContext.setBytes(BUFFER_SIZE);
```

Comment:
I just double checked with @martint, and we both feel that tracking buffers is important when:
1. the buffers are held for a long time
2. there are unbounded number of buffers
3. the buffer is very "large" 
For example, if this code was creating buffered output streams and then holding them open across calls (this is what the RCFile and ORC writers do). 

In this case, the life of the 4k buffer is strictly limited to the `try` scope, and the number is limited by the concurrent threads in this code.  Since each thread has a stack size of 1M by default, I'd be more concerned about that.  So, I'd drop the tracking.